libvast/src/system/node.cpp

```diff
@@ -567,13 +566,13 @@ caf::behavior node(node_actor* self, std::string name, path dir,
     };
     // Terminate the accountant first because it acts like a source and may
     // hold buffered data.
-    if (auto accountant = registry.find_by_label("accountant"))
-      schedule_teardown(std::move(accountant));
+    if (auto [accountant] = registry.find<accountant_actor>(); accountant)
+      schedule_teardown(caf::actor_cast<caf::actor>(std::move(accountant)));
     // Take out the filesystem, which we terminate at the very end.
-    auto filesystem = registry.find_by_label("filesystem");
+    auto [filesystem] = registry.find<filesystem_actor>();
```

Comment:
It feels like you could if-constexpr on the size of the `Ts` and then either return a single value or a tuple, as before. The singleton-wrapping feels a bit awkward.

Reply:
It's awkward here, but doing one-offs is just extra maintenance and makes working with it in generic contexts non-trivial. I'd rather always keep it a tuple.

Reply:
@tobim seems to agree with me here—marking this as resolved.